api/querier/v1/querier.proto

```diff
@@ -48,6 +48,7 @@ message SelectMergeStacktracesRequest {
   string label_selector = 2;
   int64 start = 3; // milliseconds since epoch
   int64 end = 4; // milliseconds since epoch
+  int64 max_nodes = 5;
```

Comment:
Not a proto expert, but with `optional` we could get a pointer to see if it has been set or not.

```suggestion
  optional int64 max_nodes = 5;
```